File: frontend/lib/api/business.ts
import { apiClient } from './client'

export interface Business {
  id: number
  name: string
  tax_id?: string
  address?: string
  phone?: string
  email?: string
  created_at: string
  updated_at: string
}

export interface UpdateBusiness {
  name?: string
  tax_id?: string
  address?: string
  phone?: string
  email?: string
}

export const businessApi = {
  getBusiness: async (): Promise<Business> => {
    const response = await apiClient.get('/business')
    return response.data
  },
  updateBusiness: async (data: UpdateBusiness): Promise<Business> => {
    const response = await apiClient.put('/business', data)
    return response.data
  },
}

File: backend/app/models/business.py
from sqlmodel import SQLModel, Field, Relationship
from typing import Optional
from datetime import datetime
from app.models.user import User


class Business(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    user_id: int = Field(foreign_key="user.id")
    name: str
    tax_id: Optional[str] = None
    address: Optional[str] = None
    phone: Optional[str] = None
    email: Optional[str] = None
    allow_staff_stock_adjustments: bool = Field(default=False)  # Allow staff to adjust stock
    show_sensitive_data: bool = Field(default=True)  # Owner toggle for sensitive data mode
    invoice_template: Optional[str] = Field(default="simple")  # simple, modern, blue
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)
    
    user: Optional[User] = Relationship(sa_relationship_kwargs={"foreign_keys": "[Business.user_id]"})

